adr/009-docker.md

```diff
@@ -6,19 +6,55 @@ Date: 2022-01-04
 
 We will use [Docker](https://docs.docker.com/get-started/overview/) for containerization of our application.
 
+- [Reference](https://docs.docker.com/get-started/overview/)
+
 ## Status
 
 Accepted.
 
 ## Context
 
 Docker is an open platform for developing, shipping, and running applications. Docker provides the ability to package and run an application in a loosely isolated environment. Some benefits of using Docker:
-- Develop your application and its supporting components using containers.
-- The container becomes the unit for distributing and testing your application.
-- When you’re ready, deploy your application into your production environment, as a container or an orchestrated service.
-- Allows for responsive deployment and scaling.
 
-[Reference](https://docs.docker.com/get-started/overview/)
+
+## Impact
+
+### Positive
+
+- **Compatibility:** Docker containers can run consistently across various environments (development, testing, production), reducing compatibility issues and making it easier to move applications between different environments.
+
+
+- **Isolation:** Each container runs in its own isolated environment, which minimizes conflicts between applications and simplifies dependency management.
+
+
+- **Scalability:** Docker supports scaling applications easily through container orchestration tools like Kubernetes, allowing for better resource management and load handling.
+
+
+- **Rapid Development:** Containers can be quickly built, tested, and deployed, accelerating the development cycle and improving productivity.
+
+
+- **Ecosystem and Community:** Docker has a large and active community, providing a wealth of resources, tools, and best practices that can help in troubleshooting and optimization.
+
+
+### Negative
+
+- **Complexity in Orchestration:** managing a large number of containers can introduce complexity, especially without proper orchestration tools like Kubernetes.
```

Comment:
Heh, I'd actually say that orchestration tools like Kubernetes can introduce unnecessary complexity if introduced too soon.  We should do a lot of thinking and second guessing before deciding to adopt Kubernetes.

Reply:
@halprin I'm going to remove the "like Kubernetes" part. Would you agree with the remainder?